cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -121,10 +124,22 @@ pub trait Environment {
     fn parent_path(&self) -> Result<PathBuf, EnvironmentError2>;
 
     /// Path to the environment definition file
-    fn manifest_path(&self) -> PathBuf;
+    ///
+    /// Implementations may use process context from [Flox]
+    /// to find or create a path to the environment definition file.
+    ///
+    /// [Environment::manifest_path] and [Environment::lockfile_path]
+    /// may be located in different directories.
+    fn manifest_path(&self, flox: &Flox) -> Result<PathBuf, EnvironmentError2>;
```

Comment:
nonblocking note: I wonder if at some point we'll want to drop `manifest_path` and `lockfile_path` in favor of `core_environment` or something. If we have more files than just the manifest and the lock, I think we'll want to pass around the whole directory, not paths to individual files (which may happen to be checked out in separate tempdirs 🙈)

Reply:
yeah that makes sense
effectively these methods both try to dig up the care environment anyway and return paths to it